### PR TITLE
Certification emails

### PR DESCRIPTION
**Pull request by Remg**
[ch11609]

### CODE REVIEW

File: src/Adherent/Certification/CertificationRequestMessageNotifier.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+use App\Entity\CertificationRequest;
+use App\Mailer\MailerService;
+use App\Mailer\Message\CertificationRequestApprovedMessage;
+use App\Mailer\Message\CertificationRequestBlockedMessage;
+use App\Mailer\Message\CertificationRequestPendingMessage;
+use App\Mailer\Message\CertificationRequestRefusedMessage;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class CertificationRequestMessageNotifier
```
Comment (ottaviano) [inline src/Adherent/Certification/CertificationRequestMessageNotifier.php:14]: `CertificationRequestNotifier` wdyt?

File: src/Adherent/Certification/CertificationRequestMessageNotifier.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        $refusalReason = $certificationRequest->isRefusedWithOtherReason()
+            ? $certificationRequest->getCustomRefusalReason()
+            : $this->translator->trans($certificationRequest->getRefusalReason())
+        ;
+
+        $this->mailer->sendMessage(CertificationRequestRefusedMessage::create(
+            $certificationRequest,
+            $refusalReason,
+            $this->urlGenerator->generate('app_certification_request_form')
```
Comment (ottaviano) [inline src/Adherent/Certification/CertificationRequestMessageNotifier.php:50]: `..., [], [UrlGeneratorInterface::ABSOLUTE_URL]`

File: src/Mailer/Message/CertificationRequestApprovedMessage.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+{
+    public static function create(CertificationRequest $certificationRequest): self
+    {
+        $adherent = $certificationRequest->getAdherent();
+
+        return new self(
+            Uuid::uuid4(),
+            $adherent->getEmailAddress(),
+            $adherent->getFullName(),
+            'Vous êtes certifié',
```
Comment (ottaviano) [inline src/Mailer/Message/CertificationRequestApprovedMessage.php:18]: `certifié(e)`?

File: src/Mailer/Message/CertificationRequestApprovedMessage.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            [],
+            static::getRecipientVars($adherent->getFirstName())
```
Comment (ottaviano) [inline src/Mailer/Message/CertificationRequestApprovedMessage.php:20]: as an alternative without additional method:
 
```suggestion
            ['first_name' => self::escape($adherent->getFirstName())]
```

File: src/Adherent/Certification/CertificationRequestMessageNotifier.php (hunk 2 above)
Comment (Remg) [reply src/Adherent/Certification/CertificationRequestMessageNotifier.php:50]: this is the default behavior

Comment (ottaviano) [reply src/Adherent/Certification/CertificationRequestMessageNotifier.php:50]: the default is `ABSOLUTE_PATH`, no? 

image

Comment (Remg) [reply src/Adherent/Certification/CertificationRequestMessageNotifier.php:50]: yes indeed \o/

**Pull request merged**